cpp/src/join/conditional_join.cu

```diff
@@ -36,6 +36,101 @@
 namespace cudf {
 namespace detail {
 
+std::unique_ptr<rmm::device_uvector<size_type>> conditional_join_anti_semi(
+  table_view const& left,
+  table_view const& right,
+  ast::expression const& binary_predicate,
+  join_kind join_type,
+  std::optional<std::size_t> output_size,
+  rmm::cuda_stream_view stream,
+  rmm::mr::device_memory_resource* mr)
+{
+  auto right_num_rows{right.num_rows()};
+  auto left_num_rows{left.num_rows()};
```

Comment:
Nit: Is it expensive to call `num_rows()`? If not, these variables don't save anything and decrease readability (less locality of reference). They don't save much line length either (2 chars).

Reply:
This is mostly preference on my side but I doubt it will be expensive. @bdice @PointKernel  Please comment if I am wrong.

Reply:
It is not expensive to call `num_rows()`. I don't have a strong preference on this, it's fine with me either way.

If you do have a variable, make it const.
```suggestion
  auto const right_num_rows{right.num_rows()};
  auto const left_num_rows{left.num_rows()};
```

Reply:
The reason I say it decreases readability is I had to go and look where it is assigned to know what it actually holds. Whereas `right.num_rows()` shows me clearly that it is a call to the `num_rows()` method of the `right` table.

Reply:
Fair, I can definitely empathize with preferring more logic to be inline. I'll change it to get rid of the variables. Tbh I just tend to lean towards variables for the (minuscule) performance as a habit.

Reply:
@harrism Do you think the use of variables in the actual kernels themselves are fine? I can change those over as well if not. 

@bdice Did suggest to keep consistent among both files which I agree with. For now I'll address these functions and if need be I can update the kernels as well.